### PR TITLE
Fix unit entity

### PR DESCRIPTION
**Pull request by theQuazz**

### CODE REVIEW

File: wurst/systems/units/UnitEntity.wurst [hunk 1]
```diff
@@ -32,8 +32,8 @@ public abstract class UnitEntity
     static function register(int unitTypeId, Function<unit, thistype> factory)
         factories.put(unitTypeId, factory)
 
-    static function findForUnit(unit hostile) returns thistype
-        return instances[hostile.getIndex()]
+    static function findForUnit(unit whichUnit) returns thistype
```
Comment (jonathanmclaus) [inline wurst/systems/units/UnitEntity.wurst]: Why not just `u`? This comes up enough that a standardized name would be good.

File: wurst/systems/units/hostiles/Hostile.wurst [hunk 2]
```diff
@@ -36,9 +37,6 @@ public abstract class Hostile extends UnitEntity
         for i = 1 to numCorpses
             createCorpse(players[PLAYER_NEUTRAL_AGGRESSIVE], UNIT_MEAT, pos, randomAngle())
 
-        let iter = drops.iterator()
-        for drop from iter
-            createItem(drop, pos)
-        iter.close()
+        drops.each(drop -> createItem(drop, pos))
```
Comment (jonathanmclaus) [inline wurst/systems/units/hostiles/Hostile.wurst]: Do you need to destroy `drops` after this?

File: wurst/systems/units/UnitEntity.wurst (hunk 1 above)
Comment (theQuazz) [reply wurst/systems/units/UnitEntity.wurst]: `whichUnit` is also standard, that is what they use in common.j and throughout the stdlib

File: wurst/systems/units/hostiles/Hostile.wurst (hunk 2 above)
Comment (theQuazz) [reply wurst/systems/units/hostiles/Hostile.wurst]: `.each` will cleanup the list unless it is owned (which this list is not)

**Pull request merged**